scripts/ci/install-dev-deps

```diff
@@ -25,4 +26,7 @@ def run(command):
 
 if __name__ == "__main__":
     with cd(REPO_ROOT):
-        run("pip install -r requirements-dev-lock.txt")
+        if sys.version_info[:2] == (3, 12):
+            run("pip install setuptools")
+
+    run("pip install -r requirements-dev-lock.txt")
```

Comment:
I see that because we specify the python version for `setuptools` in `requirements-dev.txt`, the dependency isn't added to the lock file. So I understand the need to be calling it separately here. However, should we be maintaining this behavior in two different places? Is there any way we can get around this? If not at the very least I think we should be pinning the version to `67.8.0` here as well.

Also any particular reason we're moving the older pip install command outside of the context manager?

Reply:
The version doesn't matter for CI and we generate the lock file for 3.7 currently, soon 3.8, which ignores the other rule. We specifically don't want to maintain the version in two places which is why it's not included here. The context manager issue was already fixed in [3b493d0](https://github.com/boto/botocore/pull/2967/commits/3b493d03cb2e0a78fc008790c22dc02fe1841771).